src/core/loss_detection.c

```diff
@@ -453,6 +460,38 @@ QuicLossDetectionOnPacketSent(
             &Connection->CongestionControl, SentPacket->PacketLength);
     }
 
+    uint64_t SendPostedBytes = Connection->SendBuffer.PostedBytes;
+
+    CXPLAT_LIST_ENTRY* Entry = Connection->Send.SendStreams.Flink;
+    QUIC_STREAM* Stream =
+        (Entry != &(Connection->Send.SendStreams)) ?
+          CXPLAT_CONTAINING_RECORD(Entry, QUIC_STREAM, SendLink) :
+          NULL;
+    
+    if (SendPostedBytes < Path->Mtu &&
+        QuicCongestionControlCanSend(&Connection->CongestionControl) &&
+        !QuicCryptoHasPendingCryptoFrame(&Connection->Crypto) &&
+        (Stream && QuicStreamAllowedByPeer(Stream)) && !QuicStreamCanSendNow(Stream, FALSE)) {
```

Comment:
I feel like a comment explaining why this logic means it is app limited. I'm not convinced this is correct.

Reply:
I'll add a comment later- it's the "rate estimation" spec that is weirdly separate from the main BBR spec. I'll double check the logic too.